### PR TITLE
[JENKINS-73103] Prepare plugin to  allow users with overall/manage to manage clouds/pod/container templates

### PR DESCRIPTION
**Pull request by mikecirioli**
[JENKINS-73103](https://issues.jenkins.io/browse/JENKINS-73103)

Initial work to  allow users with `Overall/Manage` to configure clouds (and pod/container templates).  This PR is focused on adding `Overall/Manage` support for kubernetes clouds and pod templates.  A separate PR against jenkins core is needed in order to make the clouds `management-link` visible.
 
### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStep.java [hunk 1]
```diff
@@ -450,7 +450,7 @@ public ListBoxModel doFillCloudItems() {
             ListBoxModel result = new ListBoxModel();
             result.add("—any—", "");
             // TODO track use of SYSTEM_READ and/or MANAGE in GlobalCloudConfiguration
```
Comment (jglick) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStep.java:452]: Is this comment still relevant?

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/events.jelly [hunk 2]
```diff
@@ -24,7 +24,7 @@ THE SOFTWARE.
 
 <?jelly escape-by-default='true'?>
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:i="jelly:fmt">
-  <l:layout title="${it.displayName} log" permission="${app.ADMINISTER}">
+  <l:layout title="${it.displayName} log" permission="${app.MANAGE}">
```
Comment (jglick) [inline src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/events.jelly:27]: Unrelated to PR title I guess, along with `kubectl` wrapper and other stuff.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStep.java (hunk 1 above)
Comment (mikecirioli) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStep.java:452]: not sure, i will check

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/events.jelly (hunk 2 above)
Comment (mikecirioli) [reply src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer/events.jelly:27]: i've updated the PR title know that i have a better understanding this plugin in general

**Comment by Vlatombe** [comment]
Relevant CI failure
```
Caused by: org.xml.sax.SAXParseException; systemId: file:/home/jenkins/agent/workspace/lugins_kubernetes-plugin_PR-1546/target/classes/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud/new.jelly; lineNumber: 39; columnNumber: 33; The prefix ":l" for element ":l:hasAdministerOrManage" is not bound.
	at java.xml/com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException(ErrorHandlerWrapper.java:204)
```

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStep.java (hunk 1 above)
Comment (mikecirioli) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStep.java]: AFAICT its not, will remove

**Comment by amuniz** [comment]
> A separate PR against jenkins core is needed in order to make the clouds management-link visible

Was this not needed finally?

**Pull request merged**